cmd/main.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/google/go-github/github"
+)
+
+var errorPlatformNotSupported error = errors.New("platform not supported, only supporting windows and drawin")
+
+func IsInstalled() (bool, error) {
+	execFilePath, execFileErr := getExecutableFile()
+
+	if execFileErr != nil {
+		return false, execFileErr
+	}
+
+	return fileExists(execFilePath)
+}
+
+func fileExists(filePath string) (bool, error) {
+	if _, err := os.Stat(filePath); err == nil {
+		return true, nil
+	} else if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	} else {
+		return false, err
+	}
+}
+
+func Install() error {
+	assetDownloadPath := downloadLatestReleaseAsset()
+	defer os.Remove(assetDownloadPath)
+	switch runtime.GOOS {
+	case "darwin":
+		// Use native unzip tool as it handles the creation of required symbolic links
+		cmd := exec.Command("unzip", assetDownloadPath, "-d", "/Applications")
+
+		if out, err := cmd.CombinedOutput(); err != nil {
```

Comment:
I think it would be better to use Run and return the error in case command fails to run 
https://pkg.go.dev/os/exec#Cmd.Run